### PR TITLE
Spar tests cleanup

### PR DESCRIPTION
**Pull request by fisx**
Depends on https://github.com/wireapp/saml2-web-sso/pull/63

### CODE REVIEW

File: services/spar/test-integration/Test/Spar/APISpec.hs [hunk 1]
```diff
@@ -849,7 +851,7 @@ specCRUDIdentityProvider = do
         it "responds with 2xx; makes IdP available for GET /identity-providers/" $ do
           env <- ask
           (owner, _) <- call $ createUserWithTeam (env ^. teBrig) (env ^. teGalley)
-          metadata <- Aeson.encode . (IdPMetadataValue mempty) . fst <$> makeTestIdPMetadata
+          (SampleIdP (Aeson.encode . (IdPMetadataValue mempty) -> metadata) _ _ _) <- makeSampleIdPMetadata
```
Comment (mheinzel) [inline services/spar/test-integration/Test/Spar/APISpec.hs:854]: Using the field accessor could avoid view patterns.

File: services/spar/test-integration/Test/Spar/APISpec.hs [hunk 2]
```diff
@@ -549,14 +549,14 @@ specCRUDIdentityProvider = do
...
-        passes
+        liftIO passes
     context "known IdP, client is team owner (authenticated via sso, user without email)" $ do
       it "responds with 2xx and IdP" $ do
         env <- ask
         (firstOwner, tid, idp, (_, privcreds)) <- registerTestIdPWithMeta
         ssoOwner <- mkSsoOwner firstOwner tid idp privcreds
         _ <- call $ callIdpGet (env ^. teSpar) (Just ssoOwner) (idp ^. idpId)
-        passes
+        liftIO passes
```
Comment (mheinzel) [inline services/spar/test-integration/Test/Spar/APISpec.hs:559]: using `void` also works.

File: services/spar/test-integration/Test/Spar/APISpec.hs (hunk 1 above)
Comment (fisx) [reply services/spar/test-integration/Test/Spar/APISpec.hs]: b97a104

File: services/spar/test-integration/Test/Spar/APISpec.hs (hunk 2 above)
Comment (fisx) [reply services/spar/test-integration/Test/Spar/APISpec.hs]: b97a104

**Pull request merged**